[Doc] Graph Construction PR Refactor - PR1: GSProcessing Doc Refactor (#907)
awslabs/graphstorm: docs/source/graph-construction/gs-processing/prerequisites/developer-guide.rst

> .. _developer_guide:

Reviewer: ditto for specific

> Developer Guide

Reviewer 2: Please add a short introduction of what this doc is providing.
BTW, I do not see any link from index.rst to this doc.

Author: I do add a short description for this doc. A brief clarification for this doc: we only expect this doc will be read by the developer as it is a developer guidance for anyone who wants to make contribution to the gsprocessing. So we do not add it to the index tree in the index.rst file. But we provide a reference in the gs-processing-getting-started.rst for any future developer.

Reviewer 2: Then you should put some more information in the intro.
What is the purpose of this doc? For example, for developers to follow when contributing to GraphStorm.
What is included in this doc? For example, how to setup dev env, how to apply lint check, unitest, etc.

Author: Agree. Update the introduction here to include more information

.. _gsprocessing_developer_guide:

Developer Guide
---------------

> The document helps developers set up their environment and start contributing correctly.

Reviewer 2: ```suggestion
The document helps developers set up their environment for development.
```
